task: #1768 stream progress per tool

Cerchie opened the pull request:
## Summary of Changes

Addresses #1768. 

Trying to think about what would be helpful to send in `result` besides content length 🤔 

## Any additional details or context that should be provided?


## Pull request checklist

Please check if your PR fulfills the following (if applicable):

##### Tests

- [ ] Added new
- [ ] Updated existing
- [ ] Deleted existing

##### Other

- [ ] All new disposables (event listeners, views, channels, etc.) collected as  for eventual cleanup?
<!-- prettier-ignore -->
- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md) or [README](https://github.com/confluentinc/vscode/blob/main/public/README.md)?
- [ ] Have you validated this change locally by [packaging](https://github.com/confluentinc/vscode/blob/main/README.md#packaging-steps) and installing the extension `.vsix` file?
  ```shell
  gulp clicktest
  ```

shouples commented on src/chat/tools/listTemplates.ts:
```diff
@@ -74,6 +74,7 @@ export class ListTemplatesTool extends BaseLanguageModelTool<IListTemplatesParam
       },
       token,
     );
+    stream.progress(`Found ${result.content.length} project templates...`);
```
```suggestion
    stream.progress(`Found ${result.content.length} project templates.`);
```

shouples commented on src/chat/tools/base.ts:
```diff
@@ -19,8 +19,6 @@ import { LanguageModelToolContribution } from "./types";
 export abstract class BaseLanguageModelTool<T> implements LanguageModelTool<T> {
   abstract readonly name: string;
-  /** Message to be shown when this tool is called in a chat session via `stream.progress()`. */
-  abstract readonly progressMessage: string;
```
Don't forget to remove the `readonly progressMessage` bits from the existing tools as well!

Cerchie commented:
thanks for the review! I just

- [x] parsed the object in the result of CreateProject 
- [x] added a resultCount property to getTemplate
- [x] corrected the typo in the listTemplates progress msg
- [x] removed the `readonly progressMessage` bits
- [x] added input info to the initial progress messages 

I'm encountering a 400 error with getTemplate right now, going to address that

shouples commented on src/chat/tools/getDockerContainers.ts:
```diff
@@ -121,6 +120,10 @@ export class GetDockerContainersTool extends BaseLanguageModelTool<IGetDockerCon
     const parameters = toolCall.input as IGetDockerContainersParameters;
 
+    stream.progress(
+      `Retrieving available Docker containers with parameters: ${JSON.stringify(parameters)}...`,
```
Since `resourceKind` is required, we should be able to reference it directly here:
```suggestion
      `Retrieving available ${parameters.resourceKind} Docker containers...`,
```

shouples commented on src/chat/tools/getConnections.ts:
```diff
@@ -133,6 +139,9 @@ export class GetConnectionsTool extends BaseLanguageModelTool<IGetConnectionsPar
       },
       token,
     );
+    stream.progress(
+      `Found ${result.content.length} connections for connectionType: ${parameters.connectionType}.`,
+    );
```
Let's add another `if (!parameters.connectionType)` check here in the event the model doesn't pass one, so we can keep the progress message generic instead of showing `...for connectionType: undefined`
